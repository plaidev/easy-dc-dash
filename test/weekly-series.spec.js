import Vue from 'vue';
import { mount, createLocalVue } from 'vue-test-utils';
import CssModuleTestHelperMixin from './helpers/css-modules-test-helper-mixin';

import Store from '../libs/store';
import WeeklySeries from '@/weekly-series.vue';

const localVue = createLocalVue();
localVue.use(CssModuleTestHelperMixin);

describe('weekly-series', () => {
  Store.registerData([
    { date: new Date('2017-05-18 10:00'), d1: 'a', d2: true, v1: 1, v2: 10 },
    { date: new Date('2017-05-19 10:00'), d1: 'a', d2: false, v1: 2, v2: 20 },
    { date: new Date('2017-05-20 10:00'), d1: 'b', d2: false, v1: 3, v2: 30 },
    { date: new Date('2017-05-21 10:00'), d1: 'b', d2: true, v1: 5, v2: 50 }
  ]);

  const wrapper = mount(localVue.extend(WeeklySeries), {
    localVue,
    propsData: {
      id: 'test-weekly-series',
      dimension: 'd.date',
      reduce: 'v'
    },
    attachToDocument: true
  });

  it('to match snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
