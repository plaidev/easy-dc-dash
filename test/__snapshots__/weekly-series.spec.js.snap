// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`weekly-series to match snapshot 1`] = `
<div class="outer-container" style="width: 377px; height: 233px;">
    <div class="backdrop"></div>
    <div class="card-container">
        <div class="inner-container" style="width: 377px; height: 233px;">
            <div class="container-header">
                <div class="icon-box">
                    <i class="fa fa-window-maximize"></i>
                </div>
                <!---->
            </div>
            <div class="render-area">
                <div id="test-weekly-series" class="krt-dc-component dc-chart" style="display: flex; position: relative; width: 100%; height: 100%;">
                    <!---->
                    <div class="reset-button">
                        <a class="reset" style="display: none;">
                            <i class="fa fa-ban">&nbsp;</i> <span>フィルター解除</span></a>
                    </div>
                    <!---->
                    <svg width="200" height="200">
                        <g>
                            <g class="grid-line horizontal" transform="translate(40,20)"></g>
                            <g class="grid-line vertical" transform="translate(40,20)">
                                <line x1="7.5" y1="150" x2="7.5" y2="0" opacity="0"></line>
                                <line x1="22.5" y1="150" x2="22.5" y2="0" opacity="0"></line>
                                <line x1="37.5" y1="150" x2="37.5" y2="0" opacity="0"></line>
                                <line x1="52.5" y1="150" x2="52.5" y2="0" opacity="0"></line>
                                <line x1="67.5" y1="150" x2="67.5" y2="0" opacity="0"></line>
                                <line x1="82.5" y1="150" x2="82.5" y2="0" opacity="0"></line>
                                <line x1="97.5" y1="150" x2="97.5" y2="0" opacity="0"></line>
                            </g>
                            <g class="chart-body" transform="translate(40, 20)" clip-path="url(about:blank#test-weekly-series-clip)"></g>
                            <g class="sub _0">
                                <g class="chart-body" transform="translate(40, 20)" clip-path="url(about:blank#test-weekly-series-clip)">
                                    <g class="stack-list">
                                        <g class="stack _0">
                                            <path class="line" stroke="#1f77b4"></path>
                                        </g>
                                    </g>
                                    <g class="dc-tooltip-list">
                                        <g class="dc-tooltip _0">
                                            <path class="yRef" style="display: none;" stroke-dasharray="5,5"></path>
                                            <path class="xRef" style="display: none;" stroke-dasharray="5,5"></path>
                                            <circle class="dot" r="5" style="fill-opacity: 0.000001; stroke-opacity: 0.000001;" fill="#1f77b4"></circle>
                                            <circle class="dot" r="5" style="fill-opacity: 0.000001; stroke-opacity: 0.000001;" fill="#1f77b4"></circle>
                                            <circle class="dot" r="5" style="fill-opacity: 0.000001; stroke-opacity: 0.000001;" fill="#1f77b4"></circle>
                                        </g>
                                    </g>
                                </g>
                            </g>
                            <g class="sub _1">
                                <g class="chart-body" transform="translate(40, 20)" clip-path="url(about:blank#test-weekly-series-clip)">
                                    <g class="stack-list">
                                        <g class="stack _0">
                                            <path class="line" stroke="#ff7f0e"></path>
                                        </g>
                                    </g>
                                    <g class="dc-tooltip-list">
                                        <g class="dc-tooltip _0">
                                            <path class="yRef" style="display: none;" stroke-dasharray="5,5"></path>
                                            <path class="xRef" style="display: none;" stroke-dasharray="5,5"></path>
                                            <circle class="dot" r="5" style="fill-opacity: 0.000001; stroke-opacity: 0.000001;" fill="#ff7f0e"></circle>
                                        </g>
                                    </g>
                                </g>
                            </g>
                            <g class="axis x" transform="translate(40,170)">
                                <g class="tick" style="opacity: 0.000001;" transform="translate(15,0)">
                                    <line y2="6"></line>
                                    <text dy=".71em" style="text-anchor: middle;" y="9">Sun</text>
                                </g>
                                <g class="tick" style="opacity: 0.000001;" transform="translate(30,0)">
                                    <line y2="6"></line>
                                    <text dy=".71em" style="text-anchor: middle;" y="9">Mon</text>
                                </g>
                                <g class="tick" style="opacity: 0.000001;" transform="translate(45,0)">
                                    <line y2="6"></line>
                                    <text dy=".71em" style="text-anchor: middle;" y="9">Tue</text>
                                </g>
                                <g class="tick" style="opacity: 0.000001;" transform="translate(60,0)">
                                    <line y2="6"></line>
                                    <text dy=".71em" style="text-anchor: middle;" y="9">Wed</text>
                                </g>
                                <g class="tick" style="opacity: 0.000001;" transform="translate(75,0)">
                                    <line y2="6"></line>
                                    <text dy=".71em" style="text-anchor: middle;" y="9">Thu</text>
                                </g>
                                <g class="tick" style="opacity: 0.000001;" transform="translate(90,0)">
                                    <line y2="6"></line>
                                    <text dy=".71em" style="text-anchor: middle;" y="9">Fri</text>
                                </g>
                                <g class="tick" style="opacity: 0.000001;" transform="translate(105,0)">
                                    <line y2="6"></line>
                                    <text dy=".71em" style="text-anchor: middle;" y="9">Sat</text>
                                </g>
                                <path class="domain"></path>
                            </g>
                            <g class="axis y" transform="translate(40,20)">
                                <path class="domain"></path>
                            </g>
                        </g>
                        <defs>
                            <clippath id="test-weekly-series-clip">
                                <rect width="120" height="150" transform="translate(-0, -0)"></rect>
                            </clippath>
                        </defs>
                    </svg>
                </div>
            </div>
        </div>
    </div>
</div>
`;
